clients/ui/frontend/src/app/pages/modelCatalog/screens/ModelDetailsPage.tsx

```diff
@@ -58,15 +59,21 @@ const ModelDetailsPage: React.FC = () => {
   );
 
   const registerModelButton = () => {
-    if (
-      !modelRegistriesLoaded ||
-      modelRegistriesLoadError ||
-      !artifactLoaded ||
-      artifactsLoadError
-    ) {
+    if (!modelRegistriesLoaded || modelRegistriesLoadError) {
       return null;
     }
 
+    if (artifactsLoadError) {
+      return registerButtonPopover(
+        'Unable to load model artifacts',
+        'Model registration is unavailable due to an error loading model artifacts. Please try again later.',
+      );
+    }
+
+    if (!artifactLoaded) {
+      return <Spinner size="xl" />; // Still loading artifacts
```

Comment:
@manaswinidas @ppadti In the future we can render the button but use `isDisabled` and `isLoading` to make it show a spinner in the button. But we don't need to hold up this fix and I'm not sure if it works with the MUI theme. https://www.patternfly.org/components/button#progress-indicators